go/example.go

```diff
@@ -96,47 +90,55 @@ func main() {
 	client, err := flight.NewClientWithMiddleware(
 		net.JoinHostPort(config.Host, config.Port),
 		nil,
-		[]flight.ClientMiddleware{
-			flight.NewClientCookieMiddleware(),
-		},
+		[]flight.ClientMiddleware{flight.NewClientCookieMiddleware()},
 		grpc.WithTransportCredentials(creds),
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer client.Close()
 
+	flightClient := &implementations.GoFlightClient{Client: client}
```

Comment:
1. What concerns me is: we are not doing the example easier. We are adding just 1 extra wrapper solely for mocking in tests. And to understand it a reader needs to go and open 2 more files in 2 diff directories. If there is no other option, lets comment it explicitly here at least, that it is just a thin wrapper to use for testing with the same methods.
2. Naming: both client and flightClient are actually flight clients, what distinguish them is a fact of wrapping. So maybe pairs: rawClient -> client/flightClient. Actually naming something wrappedClient and using it all the way down will not add much sense to the reader.
3. Inconsistency: why we are not defer the wrapped client?

Reply:
1. From what I understand, in Go, we cannot directly mock a concrete type like flight.Client without wrapping it in an interface. The testing frameworks (gomock, testify) we use also rely on interfaces to generate mocks so this seems to be the recommended way. I added a comment to explain this.
2. Good point with the names. Now I'm just struggling with new names. WrappedClient might be the most accurate but not sure it makes yeah. What do you think of abstractClient or testableClient, and then TestableFlightClient for the implementation of the FlightClientInterface?
3. It made sense to me because flightClient is just a wrapper, and its the actual owner of the resources. Putting closing the connection under the wrapped layer seemed unnecessary

Reply:
1 Looks like flight.Client is already an interface https://github.com/apache/arrow-go/blob/main/arrow/flight/client.go#L64 ?
Can we mock it and so the non-test path will not need any other abstractions?

2 I am not sure about abstractClient. maybe just client? or testableClient?

3. Yes, but inside the run you defer methods of passed client.